### ISSUE TITLE
"Cannot read property '0' of undefined" on repeated Vector.appendAll

### ISSUE THREAD

**Issue by bwbuchanan** [emmanueltouzery/prelude.ts#17]
This code:

```typescript
import { Vector } from 'prelude.ts';

let v = Vector.ofIterable(Array(15).fill(1));

for (let i = 0; i < 100; i += 1) {
  console.log(i);
  v = v.appendAll(Array(1000).fill(1));
}
```

Will fail after 86 iterations with:

```
TypeError: Cannot read property '0' of undefined
    at Vector.getLastNode (node_modules/prelude.ts/src/Vector.ts:279:31)
    at Vector.appendAllArrays (node_modules/prelude.ts/src/Vector.ts:457:34)
    at Vector.appendAll (node_modules/prelude.ts/src/Vector.ts:419:25)
```

**Comment by bwbuchanan** [comment]
Simpler test case:

```typescript
import { Vector } from 'prelude.ts';

Vector.ofIterable(Array(86015).fill(1)).appendAll([1]);
```

**Comment by emmanueltouzery** [comment]
Thank you for the report! I'll look into this tonight. You're not really lucky with prelude bugs.. note that I just added a fuzzer test on master exactly to detect such issues.

**Comment by emmanueltouzery** [comment]
I released 0.7.13 with the fix! Thank you for the patience and the detailed bug reports!
